### PR TITLE
[Merged by Bors] - feat(AtTopBot/Archimedean): weaken TC assumptions

### PR DESCRIPTION
**Pull request by urkud**
... from linear ordered archimedean ring to an ordered archimedean ring.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by urkud** [comment]
!bench

**Comment by urkud** [comment]
The speedup can be a rounding error or the result of easier defeq tests (no need to test defeq of all `LinearOrder` fields).

File: Mathlib/Order/Filter/AtTopBot/Archimedean.lean [hunk 1]
```diff
@@ -138,24 +138,23 @@ theorem atTop_hasAntitoneBasis_of_archimedean [OrderedSemiring R] [Archimedean R
...
--- Porting note (#11215): TODO: generalize to a `StrictOrderedRing`
-theorem atBot_hasCountableBasis_of_archimedean [LinearOrderedRing R] [Archimedean R] :
-    (atBot : Filter R).HasCountableBasis (fun _ : ℤ => True) fun m => Iic m :=
-  { countable := to_countable _
-    toHasBasis :=
-      atBot_basis.to_hasBasis
-        (fun x _ => let ⟨m, hm⟩ := exists_int_lt x; ⟨m, trivial, Iic_subset_Iic.2 hm.le⟩)
-        fun m _ => ⟨m, trivial, Subset.rfl⟩ }
+theorem atBot_hasCountableBasis_of_archimedean [OrderedRing R] [Archimedean R] :
+    (atBot : Filter R).HasCountableBasis (fun _ : ℤ => True) fun m => Iic m where
```
Comment (grunweg) [inline Mathlib/Order/Filter/AtTopBot/Archimedean.lean:146]: You're welcome to use \mapsto instead of =>, per the style guide: this is completely optional.

**Review by grunweg — COMMENTED** [review]
Thanks!
maintainer merge

**Comment by kim-em** [comment]
bors merge

**Pull request closed without merging**